ares/n64/cpu/cpu.cpp

```diff
@@ -35,6 +35,16 @@ auto CPU::main() -> void {
 
 auto CPU::step(u32 clocks) -> void {
   Thread::clock += clocks;
+
+  clocks >>= 1;
+  i33 dist = scc.compare - scc.count;
+  if(0 <= dist && dist < clocks){
```

Comment:
Is there any correctness benefit to checking the compare condition here rather than instructionEpilog()? The interrupt won't be handled until the beginning of the next instruction. I'm concerned about the performance implications of potentially redundant work being performed on every call to step().

Performing the same check in instructionEpilog() would require knowing the cycle count at the start of the instruction, which could be stored in a field.

Reply:
It doesn't sound like there's a good place where to record the cycle count at the beginning of each instruction in the recompiler, isn't there?

Reply:
More precisely, the field would represent the cycle count after the last time this logic ran. The logical place to store that would be right after this code.